### PR TITLE
Harden some checks.

### PR DESCRIPTION
**Pull request by gnusi**

### CODE REVIEW

File: core/search/min_match_disjunction.hpp [hunk 1]
```diff
@@ -159,18 +159,12 @@ class min_match_disjunction : public doc_iterator,
...
 
-    /* execute seek for all lead iterators and
-     * move one to head if it doesn't hit the target */
+    // execute seek for all lead iterators and
+    // move one to head if it doesn't hit the target
     for (auto it = lead(), end = heap_.end(); it != end;) {
       assert(*it < itrs_.size());
       const auto doc = itrs_[*it]->seek(target);
 
-      if (doc == target) {
```
Comment (MBkkt) [inline core/search/min_match_disjunction.hpp:168]: This code change about check target != eof?

Comment (MBkkt) [reply core/search/min_match_disjunction.hpp:168]: Why not check it once?

**Review by Dronplane — APPROVED** [review]
LGTM

File: core/search/min_match_disjunction.hpp (hunk 1 above)
Comment (gnusi) [reply core/search/min_match_disjunction.hpp:168]: no, this checks if doc is equal to target after seek(...). The check was moved.

**Pull request merged**